src/main/scala/io/iohk/ethereum/ledger/BlockImport.scala

```diff
@@ -161,7 +165,11 @@ class BlockImport(
     reorgResult match {
       case Some(execResult) =>
         execResult.fold(
-          err => BlockImportFailed(s"Error while trying to reorganise chain: $err"),
+          {
+            case MPTError(reason) if reason.isInstanceOf[MissingNodeException] =>
+              BlockImportFailedDueToMissingNode(reason.asInstanceOf[MissingNodeException])
+            case err => BlockImportFailed(s"Error while trying to reorganise chain: $err")
+          },
```

Comment:
this could also be written as:
```scala
          {
            case MPTError(reason: MissingNodeException) => BlockImportFailedDueToMissingNode(reason)
            case err => BlockImportFailed(s"Error while trying to reorganise chain: $err")
          },
```

Reply:
Done.